src/Access/AccessControlManager.cpp

```diff
@@ -419,6 +421,18 @@ std::shared_ptr<const ContextAccess> AccessControlManager::getContextAccess(
     params.http_method = client_info.http_method;
     params.address = client_info.current_address.host();
     params.quota_key = client_info.quota_key;
+
+    /// Extract the last entry from comma separated list of X-Forwarded-For addresses.
+    /// Only the last proxy can be trusted (if any).
```

Comment:
Still it seems to be more correct to get the first address from `X-Forwarded-For` if we want to account quota for each client IP separately because the client IP is the first address in `X-Forwarded-For`.

Reply:
It will allow clients to sent arbitrary quota keys (by forging X-Forwarded-For header)... but the purpose of this feature is to ensure per-IP limits similar to `keyed_by_ip` but with presense of proxy.

Reply:
So, the goal is to "unwrap" one level of proxy.